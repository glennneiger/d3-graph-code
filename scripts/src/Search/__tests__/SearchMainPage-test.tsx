import React from 'react';
import SearchMainPage from '../SearchMainPage';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { reducers } from 'src/store/index';
import renderer from 'react-test-renderer';
import history from "src/history";
import ConnectedRouter from 'react-router-redux/ConnectedRouter';

const store = createStore(
  reducers,
  applyMiddleware(thunk),
);

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <SearchMainPage />  
        </ConnectedRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});