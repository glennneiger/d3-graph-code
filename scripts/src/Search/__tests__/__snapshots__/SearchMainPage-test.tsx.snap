// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="search-bar text-center col-12 col-sm-6 offset-sm-3"
  >
    <img
      alt="Nova Approval Logo"
      className="img-fluid"
      src="test-file-stub"
    />
    <form
      className="search-box"
      onSubmit={[Function]}
    >
      <div
        style={Object {}}
      >
        <div
          className="input-group"
        >
          <input
            aria-autocomplete="list"
            aria-describedby="inputGroup-sizing-sm"
            aria-expanded={false}
            aria-label="Large"
            autoComplete="off"
            autoFocus={false}
            className="form-control nova-input-search"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Search FDA Regulation..."
            role="combobox"
            type="text"
            value=""
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn"
              type="submit"
            >
              <i
                className="fas fa-search"
              />
            </button>
          </div>
        </div>
      </div>
    </form>
    <div
      className="mt-4"
    >
      Enter keywords to find info about regulation numbers, product codes, or specific premarket applications.
      <br />
      Click on the help button in the bottom left of the page to learn more.
    </div>
  </div>
  <div
    className="text-center col-12 col-sm-6 offset-sm-3"
  >
    <p>
      <b>
        Some example searches to try:
      </b>
    </p>
    <p>
      "
      pacemaker medtronic
      "
    </p>
    <p>
      "
      K162550
      "
    </p>
    <p>
      "
      cardiology regulation numbers
      "
    </p>
  </div>
</div>
`;
